apps/src/netsim/NetSimRouterNode.js

```diff
@@ -1238,7 +1254,8 @@ NetSimRouterNode.prototype.getNextNodeTowardAddress_ = function (address) {
 
   // Is it another node?
   var destinationNode = _.find(nodes, function (node) {
-    return address === node.getAddress();
+    return address === node.getAddress() ||
+        (node instanceof NetSimRouterNode && address === node.getAutoDnsAddress());
```

Comment:
consider using node.getNodeType() === NodeType.ROUTER, since that's what you use a few lines below